pkg/controllers/clustershedulingpolicysnapshot/controller_integration_test.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package clustershedulingpolicysnapshot
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	fleetv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+)
+
+const (
+	testCRPName      = "my-crp"
+	testSnapshotName = "my-snapshot"
+)
+
+func policySnapshot() *fleetv1beta1.ClusterSchedulingPolicySnapshot {
+	return &fleetv1beta1.ClusterSchedulingPolicySnapshot{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: testSnapshotName,
+			Labels: map[string]string{
+				fleetv1beta1.PolicyIndexLabel:      "1",
+				fleetv1beta1.IsLatestSnapshotLabel: "true",
+				fleetv1beta1.CRPTrackingLabel:      testCRPName,
+			},
+		},
+		Spec: fleetv1beta1.SchedulingPolicySnapshotSpec{
+			PolicyHash: []byte("hash"),
+		},
+	}
+}
+
+var _ = Describe("Test clusterSchedulingPolicySnapshot Controller", func() {
+	const (
+		timeout  = time.Second * 10
+		duration = time.Second * 10
+		interval = time.Millisecond * 250
+	)
+
+	var (
+		createdSnapshot = &fleetv1beta1.ClusterSchedulingPolicySnapshot{}
+	)
+
+	BeforeEach(func() {
+		fakePlacementController.ResetQueue()
+		By("By creating a new clusterSchedulingPolicySnapshot")
+		snapshot := policySnapshot()
+		Expect(k8sClient.Create(ctx, snapshot)).Should(Succeed())
+	})
+
+	Context("When creating new clusterSchedulingPolicySnapshot", func() {
+		It("Should ignore the event", func() {
+			By("By checking placement controller queue")
+			Consistently(func() bool {
+				return fakePlacementController.Key() == ""
+			}, duration, interval).Should(BeTrue(), "controller should ignore the create event and not enqueue the request into the placementController queue")
+
+			By("By deleting snapshot")
+			Expect(k8sClient.Get(ctx, types.NamespacedName{Name: testSnapshotName}, createdSnapshot)).Should(Succeed())
+			Expect(k8sClient.Delete(ctx, createdSnapshot)).Should(Succeed())
```

Comment:
Also, to avoid flakiness, it might be a good idea to check that the object is gone before moving onto the next test (which will call create again).

Reply:
I understand that the next test is a little bit different, but would it be better if we move this cleanup to its own AfterEach block?

Reply:
done